src/validation_workflow/validation.py

```diff
@@ -0,0 +1,24 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+
+from abc import ABC, abstractmethod
+
+
+class Validation(ABC):
+    base_url = "https://artifacts.opensearch.org/releases/bundle/"
+
+    def __init__(self, version: str, distribution: str, platform: str, projects: list) -> None:
+        self.version = version
+        self.distribution = distribution
+        self.platform = platform
+        self. projects = projects
+
+    @classmethod
+    @abstractmethod
+    def download_artifacts(self, projects: list, version: str, platform: str, architectures: list) -> bool:
```

Comment:
Greetings, @Divyaasm , I am working on a dependant issue/PR to validation docker artifact and want to update you that I will need only the 'version' argument to Pull and Spin-up containers by docker-compose; hence, if you can make other arguments in your script as _optional_, that will help!  Thanks!

Reply:
+1 I believe we can make platform, architectures optional